EShop.Web/Services/InMemoryRepository.cs

```diff
@@ -5,18 +5,62 @@
 
 namespace EShop.Web.Services
 {
-    public class InMemoryRepository<T> : IRepository<T>
+    public class InMemoryRepository<T> : IRepository<T> where T : IIdentifiable
     {
         private readonly List<T> entities = new List<T>();
 
+        public int lastId = 0;
+
         public void Add(T entity)
         {
+            this.lastId = this.lastId + 1;
+            entity.Id = this.lastId;
             this.entities.Add(entity);
         }
 
         public IEnumerable<T> GetAll()
         {
             return this.entities;
         }
+
+        public T GetById(int id)
+        {
+            foreach (var entity in this.entities)
+            {
+                if (entity.Id == id)
+                {
+                    return entity;
+                }
+            }
+            
+            throw new Exception("Entity not found!");       
+        }
+
+        public void Remove(int id)
+        {
+            foreach(var entity in this.entities) 
+            {
+                if (entity.Id == id)
+                {
+                    entities.Remove(entity);
+                    break;
```

Comment:
In loc de break aici, ai putea face `return;` e echivalent, dar mai clean pentru a vedea punctele de iesire din functie.

Reply:
`return;` e echivaltent la `return void;` (nimic)